src/libpiano/response.c

```diff
@@ -712,8 +955,370 @@ PianoReturn_t PianoResponse (PianoHandle_t *ph, PianoRequest_t *req) {
 			}
 			break;
 		}
-	}
 
+		case PIANO_REQUEST_GET_PLAYBACK_INFO: {
+			PianoRequestDataGetPlaylist_t *reqData = req->data;
+			PianoSong_t *song = reqData->retPlaylist;
+
+			assert (req->responseData != NULL);
+			assert (reqData != NULL);
+			assert (reqData->quality < sizeof (qualityMap)/sizeof (*qualityMap));
+			assert (song != NULL);
+
+			json_object *audioUrlMap = NULL;
+			if (!json_object_object_get_ex (result, "audioUrlMap", &audioUrlMap)) {
+				break;
+			}
+			assert (audioUrlMap != NULL);
+
+			const char *quality = qualityMap[reqData->quality];
+			json_object *umap;
+
+			json_object *jsonEncoding = NULL;
+			if (json_object_object_get_ex (audioUrlMap, quality, &umap)) {
+				assert (umap != NULL);
+				if (json_object_object_get_ex (umap, "encoding", &jsonEncoding)) {
+					assert (jsonEncoding != NULL);
+					const char *encoding = json_object_get_string (jsonEncoding);
+					assert (encoding != NULL);
+					for (size_t k = 0; k < sizeof (formatMap)/sizeof (*formatMap); k++) {
+						if (strcmp (formatMap[k], encoding) == 0) {
+							song->audioFormat = k;
+							break;
+						}
+					}
+					song->audioUrl = PianoJsonStrdup (umap, "audioUrl");
+				}
+			}
+
+			if(song->audioUrl == NULL) {
+			/* requested quality is not available */
+				LOG("quality %s not found in audioUrlMap\n",quality);
+				ret = PIANO_RET_QUALITY_UNAVAILABLE;
+				PianoDestroyPlaylist (reqData->retPlaylist);
+				goto cleanup;
+			}
+			break;
+		}
+
+		case PIANO_REQUEST_GET_USER_PROFILE: {
+			assert (req->responseData != NULL);
+
+			json_object *stations;
+			json_object *annotations = NULL;
+			ph->user.IsPremiumUser = getBoolDefault (result,"isPremiumUser", false);
+			LOG("IsPremiumUser: %d\n",ph->user.IsPremiumUser);
+
+			if (!json_object_object_get_ex (result, "annotations", &annotations)) {
+				break;
+			}
+			assert (annotations != NULL);
+			int Annotations = 0;
+			json_object_object_foreach(annotations,Key,Val) {
+				const char *type = PianoJsonGetStr(Val,"type");
+
+				Annotations++;
+				if(strcmp(type,"PL") == 0) {
+					ph->user.PlayListCount++;
+				}
+				else if(strcmp(type,"ST") == 0) {
+					ph->user.StationCount++;
+				}
+				else if(strcmp(type,"AL") == 0) {
+					ph->user.AlbumCount++;
+				}
+				else if(strcmp(type,"TR") == 0) {
+					ph->user.TrackCount++;
+				}
+				else if(strcmp(type,"LI") == 0) {
+				}
+				else if(strcmp(type,"AR") == 0) {
+				}
+				else {
+					LOG("type %s ignored\n",type);
+				}
```

Comment:
Do we actually use these values somewhere? Do we need to store them? pianobar uses linked lists for everything anyways.